diff -Naur a/project/cmake/modules/FindSmbClient.cmake b/project/cmake/modules/FindSmbClient.cmake
--- a/project/cmake/modules/FindSmbClient.cmake	2016-09-17 08:35:33.000000000 -0700
+++ b/project/cmake/modules/FindSmbClient.cmake	2016-09-25 13:38:38.969125892 -0700
@@ -14,7 +14,7 @@
 #
 #   SmbClient::SmbClient   - The SmbClient library
 
-if(PKGCONFIG_FOUND)
+if(PKG_CONFIG_FOUND)
   pkg_check_modules(PC_SMBCLIENT smbclient QUIET)
 endif()
 
